### PR TITLE
Refactor/simplify aggregate trait

### PR DESCRIPTION
**Pull request by danieleades**
**this is a breaking change**

### CODE REVIEW

File: eventually-core/src/aggregate.rs [hunk 1]
```diff
@@ -56,14 +56,12 @@ pub trait Aggregate {
     /// of [`Event`](Aggregate::Event)s to apply the
     /// [`State`](Aggregate::State) mutation based on the current representation
     /// of the State.
-    fn handle<'a, 's: 'a>(
+    fn handle<'a>(
```
Comment (ar3s3ru) [inline eventually-core/src/aggregate.rs:59]: Do we even need to specify the lifetime if it's only one, or elision is smart enough to figure that out?

**Review by ar3s3ru — COMMENTED** [review]
Thanks for the PR!

I was a bit hesitant on accepting these, considering the plan for this version is to get dropped at some point anyway, but I think we can still merge the changes and use them as base for v0.5.0

One thing I'd add is some notice on the README that the current master contains potential breaking changes and it is considered "bleeding edge".

Wdyt?

File: eventually-core/src/aggregate.rs (hunk 1 above)
Comment (danieleades) [reply eventually-core/src/aggregate.rs]: i like where your head's at! actually i took this a step further and used `async-trait` to define it.

**Comment by danieleades** [comment]
> Thanks for the PR!
> 
> I was a bit hesitant on accepting these, considering the plan for this version is to get dropped at some point anyway, but I think we can still merge the changes and use them as base for v0.5.0
> 
> One thing I'd add is some notice on the README that the current master contains potential breaking changes and it is considered "bleeding edge".
> 
> Wdyt?

I think this is all still fairly experimental, and there's likely to be a lot of API churn. I think it might be better to merge these pull requests to a `0.5.0` branch, and then when the design settles down a bit, merge that branch to `main`/`master`

it's up to you of course. I just worry some of the design decisions made might need to be revised or rolled back, and you could end up with a messy commit history

**Review by ar3s3ru — APPROVED** [review]
Looks good to me, thanks again for the PR 💯

**Comment by danieleades** [comment]
i don't really understand how this pull request can lower the code coverage, when it's reducing the total number of lines of code in the repo...
I've seen small diffs like this is some of my projects too- i *suspect* newer versions of tarpaulin are simply producing a more accurate count than the version of tarpaulin that was run against the base commit on master

**Comment by ar3s3ru** [comment]
@danieleades might be, but fluctuations are fine (here we're talking 0.03% less, like c'mon 😂)

Anyway, gonna merge this now, thanks again 🙏🏻

**Pull request merged**